File: gui/backend/models/template.py
"""Pydantic models for task templates."""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel


class TaskCategory(str, Enum):
    TRAINING = "training"
    EVALUATION = "evaluation"
    DATA_COLLECTION = "data_collection"
    SFT = "sft"
    INFERENCE = "inference"
    OTHER = "other"


# Human-readable labels for each category (Chinese).
CATEGORY_LABELS: dict[TaskCategory, str] = {
    TaskCategory.TRAINING: "训练",
    TaskCategory.EVALUATION: "评估",
    TaskCategory.DATA_COLLECTION: "数据采集",
    TaskCategory.SFT: "监督微调",
    TaskCategory.INFERENCE: "推理",
    TaskCategory.OTHER: "其他",
}


class ConfigPreset(BaseModel):
    """A named configuration preset shipped with a template."""

    id: str
    name: str
    description: str = ""
    config_path: str = ""


class TaskTemplate(BaseModel):
    id: str
    name: str
    category: TaskCategory
    description: str = ""
    entry_script: str
    config_path: str | None = None
    presets: list[ConfigPreset] = []
    env_vars: dict[str, str] = {}

File: gui/frontend/src/types/template.ts
/** A named configuration preset shipped with a template. */
export interface ConfigPreset {
  id: string;
  name: string;
  description: string;
  /** Repo-relative path to the YAML config file. */
  config_path: string;
}

/** Mirrors the backend Pydantic model for a task template. */
export interface TaskTemplate {
  id: string;
  name: string;
  category: string;
  description: string;
  /** Path to the entry script (relative to repo root). */
  entry_script: string;
  /** Config directory path (relative to repo root), if any. */
  config_path: string | null;
  /** List of config presets discovered for this template. */
  presets: ConfigPreset[];
  /** Environment variables set by the template. */
  env_vars?: Record<string, string>;
}
